### PR TITLE
Az310 handoff worker limit

### PR DESCRIPTION
**Pull request by slfritchie**
All the 'slave' business was a seriously nasty rabbit hole ... but at least now I know how to create multi-node Riak tests with EUnit.

These commits exempt handoff data from incrementing the vnode's worker limit.  The new eunit test is a bit creative about testing that, but I think that the method is sound enough.

### CODE REVIEW

**Comment by slfritchie** [comment]
Here's quicker-to-read comments from commit 2eedb1b:

Refactoring to allow unit test to verify that handoff mechanism is actually
being used when the vnode worker limit has been exceeded.
- Use `slave` to create a 2nd node.
- Refactor `prepare_runtime()` and `teardown_runtime()` to work with a slave
  node.
- Add `limits_test_()` to start a slave node, start a pipeline &
  process stuff, shut down slave, process stuff, restart slave, process
  stuff, and then verify that handoff functions were called via
  `riak_core_tracer`.
- Other minor refactorings

File: src/riak_pipe_vnode.erl [hunk 1]
```diff
@@ -1060,6 +1060,7 @@ archive_internal(#cmd_archive{fitting=F, archive=A},
                         workers_archiving=Archiving}=State) ->
     {value, Worker, NewArchiving} =
         lists:keytake(F, #worker.fitting, Archiving),
+    %% SLF TODO: ?T(Worker#fitting.mumble, [handoff], {handoff, foo, bar}),
```
Comment (beerriot) [inline src/riak_pipe_vnode.erl]: This should work if you change `Worker#fitting.mumble` to `Worker#worker.details`, I think.

Comment (slfritchie) [inline src/riak_pipe_vnode.erl]: I ought to review my diffs more carefully before committing.
That item was exploratory, before I switched to using `riak_core_tracer`
for test verification ... finding the necessary fitting info for ?T()
in all the cases I wanted wasn't easy, so I punted.

**Comment by beerriot** [comment]
+1 merge; great to have this covered

**Pull request merged**